spec/module_validator.rb

```diff
@@ -0,0 +1,167 @@
+require 'active_model'
+
+class ArrayInclusionValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    unless value.is_a?(Array)
+      record.errors.add(attribute, "#{attribute} must be an array")
+      return
+    end
+
+    if attribute == :references
+      reference_keys = []
+      value.each { |k, _v| reference_keys << k }
+      value = reference_keys
+    end
+
+    invalid_options = value - options[:in]
+    message = "contains invalid values #{invalid_options.inspect} - only #{options[:in].inspect} is allowed"
+
+    if invalid_options.any?
+      record.errors.add(attribute, :array_inclusion, message: message, value: value)
+    end
+  end
+end
+
+class ModuleValidator < SimpleDelegator
+  include ActiveModel::Validations
+
+  validate :validate_filename_is_snake_case, :validate_reference_ctx_id, :validate_author_bad_chars, :validate_target_platforms
+
+  attr_reader :mod
+
+  def initialize(mod)
+    super
+    @mod = mod
+  end
+
+  #
+  # Acceptable Stability ratings
+  #
+  VALID_STABILITY_VALUES = [
+    Msf::CRASH_SAFE,
+    Msf::CRASH_SERVICE_RESTARTS,
+    Msf::CRASH_SERVICE_DOWN,
+    Msf::CRASH_OS_RESTARTS,
+    Msf::CRASH_OS_DOWN,
+    Msf::SERVICE_RESOURCE_LOSS,
+    Msf::OS_RESOURCE_LOSS
+  ]
+
+  #
+  # Acceptable Side-effect ratings
+  #
+  VALID_SIDE_EFFECT_VALUES = [
+    Msf::ARTIFACTS_ON_DISK,
+    Msf::CONFIG_CHANGES,
+    Msf::IOC_IN_LOGS,
+    Msf::ACCOUNT_LOCKOUTS,
+    Msf::SCREEN_EFFECTS,
+    Msf::AUDIO_EFFECTS,
+    Msf::PHYSICAL_EFFECTS
+  ]
+
+  #
+  # Acceptable Reliability ratings
+  #
+  VALID_RELIABILITY_VALUES = [
+    Msf::FIRST_ATTEMPT_FAIL,
+    Msf::REPEATABLE_SESSION,
+    Msf::UNRELIABLE_SESSION
+  ]
+
+  #
+  # Acceptable site references
+  #
+  VALID_REFERENCE_CTX_ID_VALUES = %w[CVE CWE BID MSB EDB US-CERT-VU ZDI URL WPVDB PACKETSTORM LOGO SOUNDTRACK OSVDB VTS OVE]
```

Comment:
Let's split this on to new lines to make future diffing cleaner

```suggestion
  VALID_REFERENCE_CTX_ID_VALUES = %w[
     CVE
     CWE
     ... etc ...
   ]
```